### PR TITLE
remove BOM magic handling, everything is utf-8 unless stated explicitly

### PR DESCRIPTION
**Pull request by memsharded**
Close https://github.com/conan-io/conan/issues/11228

### CODE REVIEW

File: conans/test/unittests/client/util/files/load_test.py [hunk 1]
```diff
@@ -1,20 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (SSE4) [inline conans/test/unittests/client/util/files/load_test.py:1]: but you removed the test, how do we know does it still load conanfile.txt with UTF16 LE/BE?

Comment (memsharded) [reply conans/test/unittests/client/util/files/load_test.py:1]: That is the thing, I don't think that having conanfiles, or any other file to be read and processed by Conan in any random encoding is good. All Conan related files should be utf-8. 

If we think that there is some exceptional case like conanfile.txt, we might apply some logic to that specific case. What we cannot be doing is applying such BOM detection to every string and file that is read by Conan.

I will try to restore the text that throws a more clear error message if a conanfile.txt is not utf-8.

Comment (SSE4) [reply conans/test/unittests/client/util/files/load_test.py:1]: I am okay to removing the detection for network data, and files conan creates on its own.
but for files created by user (such as conanfile.txt) it should be kept IMO
we already know it happens, and we already know editors store files in different encodings. we had valid report from real users.
the test removed was testing exactly `conanfile.txt` case, I think it should be kept, and it should be kept green.
it's fine removing `decode_text` everywhere in codebase, except these places handling user-created files (`conanfile.xt`, `profile`, `settings.yml`, etc.)

Comment (memsharded) [reply conans/test/unittests/client/util/files/load_test.py:1]: Ok, I have maintained the check and allowing those encodings, just for ``conanfile.txt`` and ``profile``. Settings.yml, while allowing user changes, is kind of a Conan generated file too, lets keep it utf-8 at the moment.

**Review by lasote — APPROVED** [review]
Looks good, still a draft?

**Comment by memsharded** [comment]
> Looks good, still a draft?

I think with the latest changes keeping the detection for user files, it should be ready.

**Comment by SSE4** [comment]
> I think with the latest changes keeping the detection for user files, it should be ready.

I think it's good as well. are we missing any user-created files (maybe layout, editable, workspace, etc.?)

**Review by SSE4 — APPROVED** [review]

**Comment by memsharded** [comment]
> I think it's good as well. are we missing any user-created files (maybe layout, editable, workspace, etc.?)

Layout, editable or workspace external files do not exist anymore in 2.0, so shouldn't be an issue.

**Pull request merged**